app/scripts/modules/core/src/pipeline/config/stages/bakeManifest/bakeManifestConfig.controller.ts

```diff
@@ -0,0 +1,45 @@
+import { IController, IQService, IScope } from 'angular';
+
+import { AccountService, ExpectedArtifactService, IAccount, IExpectedArtifact } from 'core';
+
+export class BakeManifestConfigCtrl implements IController {
+  public expectedArtifacts: IExpectedArtifact[];
+  public artifactAccounts: IAccount[];
+  public templateRenderers = ['HELM2'];
+
+  constructor(public $scope: IScope,
+              $q: IQService,
+              accountService: AccountService,
+              expectedArtifactService: ExpectedArtifactService ) {
+    'ngInject';
+    if (this.$scope.stage.isNew) {
+      const defaultSelection = {
+        templateRenderer: 'HELM2',
+        expectedArtifacts: [{
+          matchArtifact: {
+            type: 'embedded/base64',
+            name: ''
+          }
+        }]
+      };
+
+      Object.assign(this.$scope.stage, defaultSelection);
+    }
+
+    $q.when(accountService.getArtifactAccounts()).then(accounts => {
```

Comment:
Isn't `accountService.getArtifactAccounts()` returning a promise? If so, you don't need `$q.when`